frontend/src/models/player.ts

```diff
@@ -172,10 +172,28 @@ export class Player {
     notifier.info("player-announced-" + announcement, { name: this.name });
   }
 
-  hasAnnounced(...announcements: Announcement[]) {
+  hasAnnounced(...announcements: Announcement[]): boolean {
     return announcements.every(a => [...this.announcements].includes(a));
   }
 
+  wasAnnounced(...announcements: Announcement[]): boolean {
+    return announcements.every(a =>
+      [...this.announcements, ...this.getTeammateAnnouncements()].includes(a)
+    );
+  }
+
+  hasTeammateAnnounced(): boolean {
+    return this.getTeammateAnnouncements().size > 0;
+  }
+
+  getTeammateAnnouncements(): Set<Announcement> {
+    return (
+      this.game?.players.find(
+        player => this.id !== player.id && this.isRe() === player.isRe()
+      )?.announcements || new Set<Announcement>()
+    );
+  }
+
```

Comment:
These methods don't seem to be related to a single player but a party of players instead. I think it might be better putting them into [party.ts](https://github.com/hamvocke/doppelkopf/blob/master/frontend/src/models/party.ts) instead.

Reply:
Oh wow. I never actually noticed party.ts 😅 
I'm using it now obviously and probably will be adding some functionality regarding silent weddings 👍🏻

`hasTeammateAnnounced()` has been removed since it's only been used for testing.
`getTeammateAnnouncements()` is actually already implemented by party.ts, i instead added a private function which passes it's result, dependign on the team you're playing in. Therefore `getParty()` has been introduced.